kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/operator/KroxyliciousOperatorBundleInstaller.java

```diff
@@ -0,0 +1,463 @@
+import org.junit.platform.commons.PreconditionViolationException;
+
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRoleBinding;
+import io.skodjob.testframe.installation.InstallationMethod;
+import io.skodjob.testframe.resources.ResourceItem;
+
+import io.kroxylicious.systemtests.Constants;
+import io.kroxylicious.systemtests.Environment;
+import io.kroxylicious.systemtests.executor.Exec;
+import io.kroxylicious.systemtests.k8s.KubeClusterResource;
+import io.kroxylicious.systemtests.resources.kubernetes.ClusterRoleBindingResource;
+import io.kroxylicious.systemtests.resources.manager.ResourceManager;
+import io.kroxylicious.systemtests.templates.kubernetes.ClusterRoleBindingTemplates;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+import io.kroxylicious.systemtests.utils.ReadWriteUtils;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+
+/**
+ * KroxyliciousOperatorBundleInstaller encapsulates the whole installation process of Kroxylicious Operator (i.e., RoleBinding, ClusterRoleBinding,
+ * ConfigMap, Deployment, CustomResourceDefinition, preparation of the Namespace). Based on the @code{Environment}
+ * values, this class installs Kroxylicious Operator using bundle yamls.
+ */
+public class KroxyliciousOperatorBundleInstaller implements InstallationMethod {
+
+    private static final Logger LOGGER = LogManager.getLogger(KroxyliciousOperatorBundleInstaller.class);
+    private static final String SEPARATOR = String.join("", Collections.nCopies(76, "="));
+
+    private static final KubeClusterResource cluster = KubeClusterResource.getInstance();
+    private static List<File> operatorFiles;
+
+    private final ExtensionContext extensionContext;
+    private String kroxyliciousOperatorName;
+    private String namespaceInstallTo;
+    private String namespaceToWatch;
+    private List<String> bindingsNamespaces;
+    private Duration operationTimeout;
+    private Duration reconciliationInterval;
+    private Map<String, String> extraLabels;
+    private final int replicas;
+
+    private String testClassName;
+    // by default, we expect at least empty method name in order to collect logs correctly
+    private String testMethodName = "";
+
+    private static final Predicate<KroxyliciousOperatorBundleInstaller> IS_EMPTY = ko -> ko.extensionContext == null && ko.kroxyliciousOperatorName == null
+            && ko.namespaceInstallTo == null &&
+            ko.namespaceToWatch == null && ko.bindingsNamespaces == null && ko.operationTimeout == null && ko.reconciliationInterval == null
+            && ko.testClassName == null && ko.testMethodName == null;
+
+    private static final Predicate<File> installFiles = file -> !file.getName().contains("Binding") && !file.getName().contains("Deployment");
```

Comment:
Why do we need the special processing for the roles and bindings?

Reply:
I'm asking this question because I don't understand what exactly you are trying to achieve.  I see lots of code doing essentially and wondering what's it all for.   Why aren't you just using `apply` to the roles and bindings resources?  You need to explain the problem you are solving so we can judge whether a) you are solving the right problem b) when you are solving the problem right.